### PR TITLE
Add check for numbers in names

### PR DESCRIPTION
**Pull request by dominikszopa**
Changes requested in DIV-970 - Disallow numerals in name fields
- Added a function to check for numbers in name fields and display an error when one is found, similar to the "Invalid date" message
- Added "name" class to most fields
- Re-ordered the checks in the Ajax call, so they aren't skipped due to a skip-Ajax condition
- Does not prevent the user from continuing, but does not save a name with a number. This is the same as the behaviour for "invalid date"
- Does not currently work on "AKA" fields, because it was messing with the other jquery code
Will check with Kevin if these changes are sufficient

### CODE REVIEW

File: edivorce/apps/core/static/js/functions.js [hunk 1]
```diff
@@ -355,6 +355,30 @@ var validateEmail = function(el){
     }
 };
 
+var validateName = function(el){
+
+    console.log('validating name')
```
Comment (charles-orc) [inline edivorce/apps/core/static/js/functions.js]: Can you clean up test code?

File: edivorce/apps/core/static/js/functions.js [hunk 2]
```diff
@@ -355,6 +355,30 @@ var validateEmail = function(el){
...
+    // because CEIS won't accept it
+    if (hasDigit.test(value)){
+        el.closest('.form-group')
+        .addClass('has-error')
+        .append('<span class="help-block">A name cannot have a number in it.</span>');
+        return false;
+    }
+
+    return true;
+}
```
Comment (charles-orc) [inline edivorce/apps/core/static/js/functions.js]: JSlint complains about missing semicolon

File: edivorce/apps/core/static/js/ajax.js [hunk 3]
```diff
@@ -5,6 +5,23 @@ var ajaxOnChange = function () {
...
+    if (el.is(".date-picker") || el.is(".date-pickers") || el.is(".date-picker-future")){
+        isValid = validateDate(el);
+    }
+
+    if (el.is("#email_textbox")){
+        isValid = validateEmail(el);
+    }
+
+    if (el.is(".name")) {
+        isValid = validateName(el);
```
Comment (charles-orc) [inline edivorce/apps/core/static/js/ajax.js]: For alias fields, you should validate name for all fields not just current field.

File: edivorce/apps/core/static/js/functions.js (hunk 1 above)
Comment (dominikszopa) [reply edivorce/apps/core/static/js/functions.js]: Whoops, missed that one, thanks. Updated in the latest commit.

File: edivorce/apps/core/static/js/functions.js (hunk 2 above)
Comment (dominikszopa) [reply edivorce/apps/core/static/js/functions.js]: Updated.

File: edivorce/apps/core/static/js/ajax.js (hunk 3 above)
Comment (dominikszopa) [reply edivorce/apps/core/static/js/ajax.js]: I'm testing any field that has a `class="name"` in it, and I had to add a `<span>` around each input field so the `has-error` class had something to attach to. The first time I tried this, it seemed to break the functionality of the aliases, but now it seems ok - not sure what I did differently. Anyhow, the change is in `alias_field.html`.

**Pull request merged**